android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java

```diff
@@ -0,0 +1,352 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.android.integrationtest;
+
+import android.os.AsyncTask;
+import android.support.annotation.Nullable;
+import com.google.protobuf.nano.MessageNano;
+import io.grpc.ChannelImpl;
+import io.grpc.stub.StreamObserver;
+import io.grpc.stub.StreamRecorder;
+import io.grpc.transport.okhttp.OkHttpChannelBuilder;
+import junit.framework.Assert;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.security.auth.x500.X500Principal;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.KeyStore;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of the integration tests, as an AsyncTask.
+ */
+public final class InteropTester extends AsyncTask<Void, Void, String> {
+  public final static String SUCCESS_MESSAGE = "Succeed!!!";
+
+  private ChannelImpl channel;
+  private TestServiceGrpc.TestServiceBlockingStub blockingStub;
+  protected TestServiceGrpc.TestService asyncStub;
+  private String testCase;
+  private TestListener listener;
+
+  private static TrustManager[] TMS;
```

Comment:
So we only need to read the test CA and store it in TMS once when the TesterActivity is created, on the App UI, every time we click a test button, a new InteropTester object will be created, and I don't want to read the test CA every time.